src/object.js

```diff
@@ -554,52 +547,71 @@ module.exports = (common) => {
           ipfs.object.put(testObj, (err, node) => {
             expect(err).to.not.exist
 
-            node.multihash((err, multihash) => {
+            ipfs.object.stat(node.multihash, (err, stats) => {
               expect(err).to.not.exist
-
-              ipfs.object.stat(multihash, (err, stats) => {
-                expect(err).to.not.exist
-                const expected = {
-                  Hash: 'QmNggDXca24S6cMPEYHZjeuc4QRmofkRrAEqVL3Ms2sdJZ',
-                  NumLinks: 0,
-                  BlockSize: 17,
-                  LinksSize: 2,
-                  DataSize: 15,
-                  CumulativeSize: 17
-                }
-                expect(expected).to.deep.equal(stats)
-                done()
-              })
+              const expected = {
+                Hash: 'QmNggDXca24S6cMPEYHZjeuc4QRmofkRrAEqVL3Ms2sdJZ',
+                NumLinks: 0,
+                BlockSize: 17,
+                LinksSize: 2,
+                DataSize: 15,
+                CumulativeSize: 17
+              }
+              expect(expected).to.deep.equal(stats)
+              done()
             })
           })
         })
 
         it('with multihash (+ Links)', (done) => {
-          const node1 = new DAGNode(new Buffer('Some data 1'))
-          const node2 = new DAGNode(new Buffer('Some data 2'))
-          node1.addNodeLink('some-link', node2, (err) => {
-            expect(err).to.not.exist
+          let node1a
+          let node1b
+          let node2
 
-            ipfs.object.put(node1, (err, node) => {
-              expect(err).to.not.exist
-              node.multihash((err, multihash) => {
+          series([
+            (cb) => {
+              DAGNode.create(new Buffer('Some data 1'), (err, node) => {
                 expect(err).to.not.exist
-                ipfs.object.stat(multihash, (err, stats) => {
-                  expect(err).to.not.exist
-                  const expected = {
-                    Hash: 'QmPR7W4kaADkAo4GKEVVPQN81EDUFCHJtqejQZ5dEG7pBC',
-                    NumLinks: 1,
-                    BlockSize: 64,
-                    LinksSize: 53,
-                    DataSize: 11,
-                    CumulativeSize: 77
-                  }
-                  expect(expected).to.deep.equal(stats)
-                  done()
-                })
+                node1a = node
+                cb()
               })
-            })
-          })
+            },
+            (cb) => {
+              DAGNode.create(new Buffer('Some data 2'), (err, node) => {
+                expect(err).to.not.exist
+                node2 = node
+                cb()
+              })
```

Comment:
Same as in Same as in https://github.com/ipld/js-ipld-resolver/pull/67#discussion_r89480770